gcovr/merging.py

```diff
@@ -240,15 +275,36 @@ def merge_function(
 
     If ``options.ignore_function_lineno`` is set,
     the two function coverage objects can have differing line numbers.
+    With following flags the merge mode can be defined:
+      - ``options.merge_function_use_line_zero``
+      - ``options.merge_function_use_line_min``
+      - ``options.merge_function_use_line_max``
+      - ``options.separate_function``
     """
     assert left.name == right.name
     if not options.ignore_function_lineno:
-        assert left.lineno == right.lineno
-
-    left.count += right.count
-
-    if options.ignore_function_lineno:
-        left.lineno = max(left.lineno, right.lineno)
+        assert left.count.keys() == right.count.keys()
```

Comment:
Instead of an assertion, consider a normal error with an error message that informs users that they can fix this with the `--merge-mode-functions` option.

Reply:
I've added a AssertionError exception.